cmd/state/main.go

```diff
@@ -33,7 +33,7 @@ import (
 	"github.com/ActiveState/cli/pkg/projectfile"
 )
 
-func main() {
+func _main() (exitCode int) {
```

Comment:
This feels very proof of concept, let's clean this up.

I kinda like the pattern I landed on with the update dialog:

http://github.com/ActiveState/cli/blob/d00dfc6951aa008c2dd8761bf88c4cd4ea399277/cmd/state-update-dialog/main.go#L17-L17

ie. put the os.Exit() in a defer itself.